src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```diff
@@ -48,14 +51,25 @@ public function isGranted($role, $object = null, $field = null)
         }
     }
 
+    public function getCsrfToken(string $tokenId): string
+    {
+        return $this->csrfTokenManager->getToken($tokenId)->getValue();
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getFunctions()
     {
-        return array(
+        $functions = array(
             new TwigFunction('is_granted', array($this, 'isGranted')),
         );
+
+        if (null !== $this->csrfTokenManager) {
+            $functions[] = new TwigFunction('csrf_token', array($this, 'getCsrfToken'));
```

Comment:
Regarding DX, wouldn't it be better to always add the method but throw if no `csrfTokenManager` injected?

Reply:
I would say it's better this way because you will notice the error when templates are parsed. Having the function fail at runtime can just go without notice.